src/lib/core/datetime/date-selection-model.ts

```diff
@@ -75,17 +78,19 @@ export class MatSingleDateSelectionModel<D> extends MatDateSelectionModel<D> {
 
   /** Gets the current selection. */
   getSelection(): D | null {
-    return this.isValid() ? this.date : null;
+    return this.isValid() ? this.adapter.deserialize(this.date) : null;
   }
 
   /**
    * Adds the given date to the selection model. For a `MatSingleDateSelectionModel` this means
    * simply replacing the current selection with the given selection.
    */
-  add(date: D) {
+  add(date: D | null, silent = false) {
```

Comment:
the thinking here is that the class doesn't actually have a common `setSelection` method as it is different for each derived class. I agree that it's a bit odd, but some components don't actually know which derived class they use, so in order to be able to set an empty date (which is valid) I allowed it. I open for other implementations, but it has to be a common one int he base class.

Reply:
Also. setting it to `null` is a correct setting. such as an empty input field.

Reply:
I think we should be able to structure it so that whenever we would be tempted to call `add(null)` we can call `setSelection(null)` instead. For example, in the input knows whether it is a range selection input or a single selection input, so it can call `setSelection`